src/test/java/de/tum/in/www1/artemis/architecture/ResourceArchitectureTest.java

```diff
@@ -34,6 +36,8 @@
 
 class ResourceArchitectureTest extends AbstractArchitectureTest {
 
+    private static final Pattern KEBAB_CASE_OR_PATH_VARIABLE_PATTERN = Pattern.compile("^(\\.?([a-z0-9]+(-[a-z0-9]+)*|\\{[^}]+\\}|[a-z0-9]+\\.json|\\*))$");
```

Comment:
I believe the outer capture group is unnecessary and the brackets could be removed.